### PR TITLE
🌱 Run a controller separately from kcp-dev/kcp

### LINKED ISSUE #59: Separated controller fails to list resources
In #58, we tried to run the `kcp-scheduling-placement-controller` separately from kcp-dev/kcp. We tracked the path along which the controller is invoked in kcp-dev/kcp, then tried to reconstruct the path in [main.go](https://github.com/kcp-dev/edge-mc/blob/0bee29baf5f930d6f6fd2a121981864ee000f247/main.go#L45-L111).

There must be something we missed in the path because the controller fails to list Placements and Locations.
Placement:
```
E1208 16:16:20.007490   72875 reflector.go:138] k8s.io/client-go@v0.0.0-20220915135949-eeba459ad2a1/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Placement: failed to list *v1alpha1.Placement: the server could not find the requested resource (get placements.scheduling.kcp.dev)
```
Locations:
```
E1208 16:16:20.043833   72875 reflector.go:138] k8s.io/client-go@v0.0.0-20220915135949-eeba459ad2a1/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Location: failed to list *v1alpha1.Location: the server could not find the requested resource (get locations.scheduling.kcp.dev)
```
The controller also watches namespaces, but it doesn't complain about listing namespaces.

We need some help to understand what are the missing parts in the path.

### LINKED ISSUE #61: Add cmd directory and subdirectory for placement controller main
We might have to host multiple binaries command, currently there is a main.go in the root directory, we should follow the convention of creating `cmd/placement/<main file>`

### PR DESCRIPTION
**Pull request by waltforme**
## Summary
This PR continues and finishes #58.

## Related issue(s)

Fixes #59
Fixes #61

### CODE REVIEW

File: cmd/placement/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+```
+into pkg/reconciler/scheduling/placement/placement_reconcile.go.reconcile(),
+and insert
+```go
+	if true {
+		fmt.Printf("%v controller skipped\n", ControllerName)
+		return false, nil
+	}
+```
+into pkg/reconciler/workload/placement/placement_reconcile.go.reconcile().
```
Comment (MikeSpreitzer) [inline cmd/placement/README.md:23]: Add "along with the required import statement".

**Comment by MikeSpreitzer** [comment]
Command line parsing is missing.  The following should produce a lot of debug log messages.

```
(base) mspreitz@mjs12 edge-mc % go run cmd/placement/main.go -v=8
I1216 14:25:21.532682   67525 controller.go:249] "Starting controller" reconciler="kcp-scheduling-placement"
```

File: cmd/placement/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```
+The status of Placement `dev` should show (1) `phase: Unbound`, (2) the `foo` Location selected by Placement `dev`.
+
+Back to the user's workspace, label namespace `default`:
+```
+kubectl label ns default env=dev
+```
+The status of the placement should show `phase: Bound`.
+
+Remove the `env=dev` label on the default namespace, then delete Location `foo`, the state machine should transit to `Unbound`, then back to `Pending`.
```
Comment (MikeSpreitzer) [inline cmd/placement/README.md:52]: I tried this demo and it worked until the very last step.  The Placement's phase retreated to Unbound and stayed there.

**Comment by waltforme** [comment]
> Command line parsing is missing. The following should produce a lot of debug log messages.
> 
> ```
> (base) mspreitz@mjs12 edge-mc % go run cmd/placement/main.go -v=8
> I1216 14:25:21.532682   67525 controller.go:249] "Starting controller" reconciler="kcp-scheduling-placement"
> ```

Thanks @MikeSpreitzer, we (You, @pdettori and myself) have consensus on that this is the next immediate thing we want to add. So Paolo created #62 and I am assigned to it. This will be handled there.

File: cmd/placement/README.md (hunk 1 above)
Comment (waltforme) [reply cmd/placement/README.md]: Fixed in a force push. Thanks @MikeSpreitzer !

**Review by MikeSpreitzer — APPROVED** [review]
/LGTM

**Pull request merged**